fabrics.c

```diff
@@ -728,8 +728,9 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 		nvme_root_set_application(r, context);
 	ret = nvme_scan_topology(r, NULL, NULL);
 	if (ret < 0) {
-		fprintf(stderr, "Failed to scan topology: %s\n",
-			 nvme_strerror(errno));
+		if (errno != ENOENT)
+			fprintf(stderr, "Failed to scan topology: %s\n",
+				nvme_strerror(errno));
```

Comment:
Sorry if I was unclear in #2241. I don't really have a problem with the warning message. The main issue is that the commands fail on `nvme-cli` 2.x where it succeeds on 1.x, which I don't think this change addresses. I think the correct fix is for `nvme_scan_topology()` not to return an error in the `ENOENT` case.

Reply:
I think it would actually be preferable to keep the error message. If the command fails, the user should know why. I posted https://github.com/linux-nvme/libnvme/pull/795 to avoid `nvme_scan_topology()` returning this error in the first place when `nvme_core` isn't loaded. So I think we can revert this change.

Reply:
And I think https://github.com/linux-nvme/libnvme/pull/795 is just the wrong thing to do